packages/web-app-webfinger/src/discovery/webfingerDiscovery.ts

```diff
@@ -0,0 +1,28 @@
+import { OwnCloudInstance } from 'web-app-webfinger/src/discovery/types'
+import { ClientService } from 'web-pkg'
+import { urlJoin } from 'web-client/src/utils'
+
+interface OwnCloudInstancesResponse {
+  subject: string
+  links: OwnCloudInstance[]
+}
+
+const OWNCLOUD_REL = 'http://webfinger.owncloud/rel/server-instance'
+
+export class WebfingerDiscovery {
+  private serverUrl: string
+  private clientService: ClientService
+
+  constructor(serverUrl: string, clientService: ClientService) {
+    this.serverUrl = serverUrl
+    this.clientService = clientService
+  }
+
+  public async discoverOwnCloudInstances(): Promise<OwnCloudInstance[]> {
+    const client = this.clientService.httpAuthenticated
+    const url =
+      urlJoin(this.serverUrl, '.well-known', 'webfinger') + `?resource=${encodeURI(this.serverUrl)}`
+    const response: OwnCloudInstancesResponse = (await client.get(url)).data
```

Comment:
fyi the request currently fails in a setup with self signed certificate, see https://github.com/owncloud/ocis/issues/6102

Reply:
doesn't fail anymore 👍